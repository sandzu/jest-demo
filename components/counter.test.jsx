import React from 'react';
import { shallow } from "enzyme";
import Counter from "./counter";
import Foo from  './foo';
import renderer from 'react-test-renderer';
// ^for snapshot testing

describe("Counter", ()=>{
  it("starts with a count of 0", () => {
    let container = shallow(<Counter/>);
    let counterCount = container.state().count;
    expect(counterCount).toBe(0);
  });

  it("state.count increments on button click", () => {
    let container = shallow(<Counter/>);
    container.find("button").simulate("click");
    let counterCount = container.state().count;
    expect(counterCount).toBe(1);
  });

  it("renders 'count' foos", () =>{
    let container = shallow(<Counter/>);
    expect(container.find("Foo").length).toBe(0);
    container.find("button").simulate("click");

    expect(container.find("Foo").length).toBe(1);
    // find() identifies components by name, not text content
  });

  it('matches the initial snaptshot', ()=>{
    const component =  renderer.create(<Counter/>);
    expect(component.toJSON()).toMatchSnapshot();
    component.getInstance().addOne();
    component.getInstance().addOne();
    expect(component.toJSON()).toMatchSnapshot();

  });
});

// still need to configure enzyme with the right container
// see configure.enzyme.js and jest.config.js, both need to be in the root
