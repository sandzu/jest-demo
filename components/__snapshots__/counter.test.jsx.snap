// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter matches the initial snaptshot 1`] = `
<div>
  <p>
     count: 
    0
  </p>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`Counter matches the initial snaptshot 2`] = `
<div>
  <p>
     count: 
    2
  </p>
  <button
    onClick={[Function]}
  />
  <h1>
      foo 
  </h1>
  <h1>
      foo 
  </h1>
</div>
`;
